ci/rpm/install_maldet.sh

```diff
@@ -0,0 +1,61 @@
+#!/bin/bash
+
+set -uex
+
+mydir="$(cd "$(dirname "${BASH_SOURCE[0]}")" >/dev/null 2>&1 && pwd)"
+# at some point we want to use: shellcheck source=ci/rpm/daos_info.sh
+# shellcheck disable=SC1091
+source "$mydir/daos_info.sh"
+
+: "{BUILD_URL:=}"
+if command -v dnf; then
+  dnf --assumeyes install clamav clamav-devel
+  if [ -n "${BUILD_URL}" ]; then
+    repo_url="${BUILD_URL}artifact/artifacts/${PUBLIC_DISTRO}${MAJOR_VERSION}/"
+    dnf --assumeyes config-manager --add-repo="$repo_url"
+    repo="${repo_url#*://}"
+    repo="${repo//%/}"
+    repo="${repo//\//_}"
+    # bug in EL7 DNF: this needs to be enabled before it can be disabled
+    dnf config-manager --save --setopt="$repo".gpgcheck=1
+    dnf config-manager --save --setopt="$repo".gpgcheck=0
+    # but even that seems to be not enough, so just brute-force it
+    if [ -d /etc/yum.repos.d ] &&
+       ! grep gpgcheck /etc/yum.repos.d/"$repo".repo; then
+        echo "gpgcheck=0" >> /etc/yum.repos.d/"$repo".repo
+    fi
+  fi
+elif command -v apt-get; then
+  apt-get --assume-yes install clamav libclamav-dev
+  service clamav-freshclam stop || true
+  systemctl disable clamav-freshclam || true
+fi
+mkdir -p /etc/clamd.d
+printf "LogSyslog yes\n" >> /etc/clamd.d/scan.conf
+
+lmd_tarball='maldetect-current.tar.gz'
+: "${JENKINS_URL:=https://build.hpdd.intel.com/}"
+
+lmd_url="${JENKINS_URL}job/daos-stack/job/tools/job/master/"
+lmd_url+="lastSuccessfulBuild/artifact/${lmd_tarball}"
+curl "${lmd_url}" --silent --show-error --fail -o "/var/tmp/${lmd_tarball}"
+
+lmd_src='lmd_src'
+mkdir -p "/var/tmp/${lmd_src}"
+tar -C "/var/tmp/${lmd_src}" --strip-components=1 -xf "/var/tmp/${lmd_tarball}"
+pushd "/var/tmp/${lmd_src}"
+  ./install.sh
+popd
+
+/usr/local/sbin/maldet --update-sigs || true
```

Comment:
how often is this check run, is it protected by a cachebust at all?

We do not want to run this on every PR, but also we do not want to silently ignore errors and run with out-of-date signatures for a week.

My preference would be to ensure this was run relatively infrequently, but for us to fail on it.  Unfortunately with docker this will cache a failure so not re-try, the logic we would want cannot be done through docker, we'd need to use Jenkins builds and check lastSuccessFullBuild to do that, but I don't think it's happening here.

Reply:
There is a CACHEBUST set before this script is run, so this section of the Dockerfile will not be cached.
 
We are going to want to run at least one malware scan per PR / branch.
The scan script will also make an attempt to update the maldet database.

As I understand it, the last outage of the maldet definition server lasted for at least 12 hours.   Because the maldet scans were disabled at that time, we do not know how long that it was actually down.

Even with out up to date maldet signatures, we are still getting up to date ClamAV database scans.

Reply:
We certainly don't need to update the scans for every PR, I'd have expected daily to be more than sufficient if we can find a way to achieve that.

Reply:
I have not found a way to cache the maldet definition file.   The choice seems to be use the one that is in the git repository as of the current tarball, which is updated periodically, or request an update for every run.

Or move the maldet install to before the CACHEBUST and let the CB0 Cache bust handle when it is updated, which is what the current push does.

Reply:
> Or move the maldet install to before the CACHEBUST and let the CB0 Cache bust handle when it is updated, which is what the current push does.

I'd missed that update but I think that would be a good solution.  I'll re-review this PR in light of this.

Reply:
This does look like it should be cached now although I am checking.  Once again can you remove the "|| true" part or explain how it's safe.  We don't want to silently ignore failure to update the signatures.

Reply:
If I do not put in the || true, if the maldet databasse is not reachable when this is updated, it will fail, and there is no recovery for it that can be done automatically.   Which would fail the job.

The main purpose of this PR is to make sure that this never causes a PR or master/release branch to fail if the maldet database is unreachable.

Which means that we may be doing scans with an older maldet database for the period of time that will be determined what is used for a CACHEBUST.

This check is mainly to give us the early alert if possible of anything that may be found, but it is not the final check that will be done before the packages are released, so it is not a problem for that period of time if the maldet database is not 100% current.

Reply:
Right now, we are not running a malware check on each build because for a few days last year the maldet database could not be reached and it was causing CI to fail.

We want to have this check run, but we also do not want CI to fail if the maldet update database goes offline again, as we have no control of it, and currently no way to mirror it, as the result would be that the scan would be shut off completely and we would be back to where we are now.

With the maldet database down, we still get the clamav current data, and we still get the maldet definitions that are in the current build of maldet, so it is not like the scan is doing nothing just because we can not update the maldet database.

And we still run additional scans on the built RPMSs before they are released.